ergotree-interpreter/src/eval/bin_op.rs

```diff
@@ -16,65 +16,77 @@ use num_traits::CheckedSub;
 use num_traits::Num;
 
 use crate::eval;
+use crate::eval::bigint::fits_in_256_bits;
 use crate::eval::env::Env;
 use crate::eval::EvalContext;
 use crate::eval::EvalError;
 use crate::eval::Evaluable;
 
+fn arithmetic_err<T: std::fmt::Display>(
+    op: &str,
+    lv_raw: T,
+    rv_raw: T,
+    err_str: &str,
+) -> EvalError {
+    EvalError::ArithmeticException(format!(
+        "({0}) {1} ({2}) resulted in {3}",
+        lv_raw, op, rv_raw, err_str
+    ))
+}
+
+fn check_bigint_overflow(res: Result<Value, EvalError>) -> Result<Value, EvalError> {
```

Comment:
Consider reducing `res` type to `Value` and use `?` before passing an argument at the call site. 
After that renaming `res` to `v` might be a better fit.

Reply:
Sounds good, just made that change and I'll push it shortly.